### PR TITLE
Connecting models

### PR DESCRIPTION
**Pull request by dwram**
### Summary

- In order to hide the site's content from unauthorised users, they are redirected if not logged in: `before_action: require_login` at the top-level of the posts controller.

- To "connect" the user model and post model, not only in the application but also at the database level, migration files were created with appropriate references, alongside complimentary has_many and belongs_to relationships in the respective .rb model files for User and Post. 

- A join table concept was implemented in User_post model.

- The post, upon instantiation, is provided access to methods to retrieve the owner's name `get_owner`.

- CSS was also implemented for the feel and look of the site, with a fixed top-navigation.
- IDs were also implemented in the views for ease of identification and querying for spec tests.
 
- After posts were submitted/updated, they would not reflect until the User reloaded the page.
Therefore, in order to communicate to the user and improve UX, in-line Javascript was implemented to force a page reload automatically and thus showing the page with the new data.

**Additional information**

This update requires a new bundle install, as Database Cleaner gem was implemented to truncate / sanitize the database after and before each successive test.

`DROP DATABASE` commands in PSQL for both `acebook_test` and `acebook_production`; recreate with `rails db:create` and `rails db:migrate`. 

P.s.s

Failing Travis CI is fixed via the Capybara implement and amendments made in the Styling PR.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
 class Post < ApplicationRecord
+  belongs_to :user
+
+  MAX_TIME_TO_UPDATE = 6
```
Comment (phump81) [inline app/models/post.rb:4]: Should this be 600?

**Review by 02ship — APPROVED** [review]

**Review by 02ship — CHANGES_REQUESTED** [review]
MAX_TIME_TO_UPDATE change to 600

File: app/models/post.rb (hunk 1 above)
Comment (dwram) [reply app/models/post.rb:4]: Been updated in #6

**Review by 02ship — APPROVED** [review]

**Comment by ralphm10** [comment]
reviewed and approved

**Pull request merged**